addon/authenticators/oauth2-password-grant.js

```diff
@@ -164,24 +161,19 @@ export default BaseAuthenticator.extend({
       }
       this.makeRequest(serverTokenEndpoint, data).then((response) => {
         run(() => {
-          if (isEmpty(response['access_token'])) {
-            throw new Error('access_token is missing in server response')
-          }
+          this._validate(reject, response, 'access_token');
           const refreshAccessTokens = this.get('refreshAccessTokens');
           if (refreshAccessTokens) {
-            if (isEmpty(response['expires_in'])) {
-              throw new Error('expires_in is missing in server response')
-            }
-            if (isEmpty(response['refresh_token'])) {
-              throw new Error('refresh_token is missing in server response')
-            }
+            this._validate(reject, response, 'expires_in');
+            this._validate(reject, response, 'refresh_token');
```

Comment:
it looks like 
```
{ error: 'invalid_grant' }
```
is returned elsewhere before `refresh_token` is validated. Maybe we can remove this?